test/e2e/node/kubelet.go

```diff
@@ -479,62 +485,266 @@ var _ = SIGDescribe("kubelet", func() {
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "--tail", n, "-o", "short")
+			result := runKubectlCommand(cmd, "--role --query --tail")
+			logs := journalctlCommand("-u", "kubelet", "-n", n, "")
+			if result != logs {
+				framework.Failf("Failed to receive the correct kubelet logs or the correct amount of lines of logs")
+			}
+		})
 
-			for nodeName := range nodeNames {
-				cmd := tk.KubectlCmd("alpha", "node-logs", nodeName, "--path", "pods")
-				runKubectlCommand(cmd, "--path")
+		/*
+			Test if kubectl node-logs --role worker --query kubelet --raw
+			returns the kubelet logs or not!
+		*/
+
+		ginkgo.It("should return the logs in the raw form for the requested query of the specific role in raw form", func() {
+			ginkgo.By("Starting the command")
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "--raw")
+			result := runKubectlCommand(cmd, "--role --query --raw")
+			lookForString("kubelet", result)
+		})
+
+		/*
+			Test if kubectl node-logs --role master --query kubelet --pattern kubelet -o short
+			returns the kubelet logs or not!
+		*/
+
+		ginkgo.It("should return the logs for the requested query of the master node and filtered by kubelet", func() {
+			ginkgo.By("Starting the command")
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "--pattern", "kubelet", "-o", "short")
+			result := runKubectlCommand(cmd, "--role --query --pattern")
+			logs := journalctlCommand("-u", "kubelet", "-b", "0", "-g", "kubelet", "")
+			if result != logs {
+				framework.Failf("Failed to receive the correct kubelet logs for the specific pattern")
+			}
+		})
+
+		/*
+			Test if kubectl node-logs --role master --query kubelet --tail 1 --output json
+			returns the kubelet logs in json format or not!
+		*/
+
+		ginkgo.It("should return the logs for the requested query of the specific node in json format", func() {
+			ginkgo.By("Starting the command")
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "--tail", "1", "--output", "json")
+			result := runKubectlCommand(cmd, "--role --query --tail --output")
+			framework.ExpectEqual(len(lines(result)), 1)
```

Comment:
this is the only use of `lines`. Does it have to be defined as a func 100 lines above?